File: app/src/main/java/com/example/xc_m6_java/manager/RealmManager.java
package com.example.xc_m6_java.manager;

import com.example.xc_m6_java.model.Post;

import java.util.ArrayList;

import io.realm.Realm;
import io.realm.RealmResults;

public class RealmManager {

    public static final String TAG = RealmManager.class.getSimpleName();
    private static RealmManager realmManager;
    private static Realm realm;

    public static RealmManager getInstance() {
        if (realmManager == null) {
            realmManager = new RealmManager();
        }
        return realmManager;
    }

    public RealmManager() {
        realm = Realm.getDefaultInstance();
    }

    public void savePost(Post post) {
        realm.beginTransaction();
        realm.copyToRealmOrUpdate(post);
        realm.commitTransaction();
    }

    public ArrayList<Post> loadPosts() {
        ArrayList<Post> posts = new ArrayList<>();
        RealmResults<Post> results = realm.where(Post.class).findAll();
        for (Post post : results) {
            posts.add(post);
        }
        return posts;
    }


}

File: app/src/main/java/com/example/xc_m6/manager/RealmManager.kt
package com.example.xc_m6.manager

import com.example.xc_m6.model.Post
import io.realm.Realm
import io.realm.RealmResults

class RealmManager {

    val TAG = RealmManager::class.java.simpleName

    companion object {
        private var realmManager: RealmManager? = null
        private  val realm: Realm

        val instance: RealmManager?
                get() {
                    if (realmManager == null) {
                        realmManager = RealmManager()
                    }
                    return realmManager
                }

        init {
            realm = Realm.getDefaultInstance()
        }
    }
    fun savePost(post: Post) {
        realm.beginTransaction()
        realm.copyToRealmOrUpdate(post) // If post that has same ID realm won't add it again. It just will update it.
        realm.commitTransaction()
    }

    fun loadPosts(): ArrayList<Post> {
        var posts: ArrayList<Post> = ArrayList()
        var results: RealmResults<Post> = realm.where(Post::class.java).findAll()
        for (post in results) {
            posts.add(post)
        }
        return posts
    }
}